examples/gallery/plot/points-categorical.py

```diff
@@ -0,0 +1,59 @@
+"""
+Color points by categories
+---------------------------
+The :meth:`pygmt.Figure.plot` method can be used to plot symbols which are
+color-coded by categories.
```

Comment:
The description here needs to be expanded a bit. It might be necessary to mention how the categorical colour coding works, i.e. using `fig.plot()` with `color=df.species.cat.codes.astype(int)` **and** a colormap via `cmap=True`.

Probably also a good idea to link to the data source https://github.com/allisonhorst/palmerpenguins.

Reply:
Made some adjustments. Thanks for the suggestions @weiji14.